Parameters can be optional

atward opened the pull request:
Addresses: #36

Makes it valid to not specify params_json or params_yaml: in which case an empty parameters list ([]) is provided.

delfick commented on bespin/option_spec/bespin_specs.py:
```diff
@@ -302,8 +284,8 @@ def stack_spec(self):
             , stack_yaml = valid_stack_yaml(default="{config_root}/{_key_name_1}.yaml")
 
-            , params_json = valid_params_json(default="{config_root}/{environment}/{_key_name_1}-params.json")
-            , params_yaml = valid_params_yaml(default="{config_root}/{environment}/{_key_name_1}-params.yaml")
```
why are you removing the defaults? if the file doesn't exist then the value gets set to NotSpecified anyway....

atward replied:
Details in PR and commit message. There's currently no situation where the default could be used due to the enforcement of one or the other. ie: the user must pass a value, and that value would be used over the NotSpecified one.
Change to these lines makes this clearer.

delfick replied:
I'm still confused.

If only the json file exists and neither are specified then params_yaml will be NotSpecified
if only the yaml file exists and neither are specified then params_json will be NotSpecified
if both the yaml and the json file exist then you have a confusing setup

if you define them in your configuration, you probably shouldn't have it also as a file somewhere.

delfick replied:
because of this line here https://github.com/atward/bespin/blob/8e3c5b856737f60a8066eadc40ff0e407402c560/bespin/option_spec/bespin_specs.py#L56 which triggers if https://github.com/atward/bespin/blob/8e3c5b856737f60a8066eadc40ff0e407402c560/bespin/option_spec/bespin_specs.py#L43 is False

atward replied:
I see it now. Will rebase PR

atward committed:
Parameters can be optional

Makes it valid to not specify params_json or params_yaml: in which case
an empty parameters list ([]) is provided.

atward commented:
Updated PR. Somehow I thought that one or the other must be specified _by the user_

delfick commented:
fair enough

delfick commented:
I've introduced https://github.com/delfick/bespin/commit/3148a674fbce381145cfc9a1f90256c44abb0e39 to fix the docs generation :)
